### DISCUSSION TITLE
Some questions about the DataLoaderDispatcherInstrumentationState.checkForNoDataLoader method

### DISCUSSION THREAD

**Discussion by PerfectBroken** [graphql-java/graphql-java#3169] [General]
I have some questions about the private boolean checkForNoDataLoader(DataLoaderRegistry dataLoaderRegistry) method in DataLoaderDispatcherInstrumentationState.java. It seems that this method cannot detect DataLoaderRegistry instances defined by the user that don't have any DataLoader.
![image](https://user-images.githubusercontent.com/30275632/224471968-eae98d23-b083-4527-886e-8dceb91f0267.png)

        <dependency>
            <groupId>com.graphql-java</groupId>
            <artifactId>graphql-java</artifactId>
            <version>17.3</version>
        </dependency>

**Comment by bbakerman** [comment]
The `DataLoaderRegistry` is a mutable class - for example `org.dataloader.DataLoaderRegistry#computeIfAbsent` allows an empty registry to be populated with new values at runtime

So the code above only detects if NO data loader registry was every set - and not if its empty

What is your concern here?
